packages/react/spec/liveQueries.spec.tsx

```diff
@@ -240,38 +302,172 @@ describe("live queries", () => {
 
-    // re-execute the live query operation
     urqlClient.reexecuteOperation(findManyOperation!);
 
-    // let the event loop run to give urql a chance to re-execute the operation
-    await sleep(1000);
+    await sleep(100);
 
-    // make sure the operation was NOT re-executed
     expect(operationCount).toBe(1);
   });
+
+  test("live queries can be re-established after mutation invalidates cache", async () => {
+    const { result } = renderHook(() => useFindMany(api.user, { live: true }), {
+      wrapper: MockGraphQLWSClientWrapper(api),
+    });
+
+    expect(result.current[0].fetching).toBe(true);
+    expect(result.current[0].data).toBeFalsy();
+
+    await waitFor(() => expect(mockGraphQLWSClient.subscribe.subscriptions).toHaveLength(1));
+
+    const subscription = mockGraphQLWSClient.subscribe.subscriptions[0];
+    subscription.push({
+      data: {
+        users: {
+          edges: [
+            {
+              node: {
+                id: "1",
+                email: "user1@test.com",
+              },
+            },
+          ],
+        },
+      },
+      revision: 1,
+    } as any);
+
+    await waitFor(() => expect(result.current[0].data!.length).toEqual(1));
+    expect(result.current[0].data![0].id).toEqual("1");
+
+    subscription.push({
+      errors: [
+        {
+          message: "Connection lost",
+          locations: [{ line: 1, column: 1 }],
+          path: ["users"],
+        },
+      ],
+      revision: 2,
+    } as any);
+
+    await waitFor(() => expect(result.current[0].error).toBeTruthy());
+  });
+
+  test("multiple live queries with different parameters work correctly", async () => {
+    const { result: result1 } = renderHook(() => useFindMany(api.user, { filter: { id: { equals: "1" } }, live: true }), {
+      wrapper: MockGraphQLWSClientWrapper(api),
+    });
+
+    const { result: result2 } = renderHook(() => useFindMany(api.user, { filter: { id: { equals: "2" } }, live: true }), {
+      wrapper: MockGraphQLWSClientWrapper(api),
+    });
+
+    expect(result1.current[0].fetching).toBe(true);
+    expect(result2.current[0].fetching).toBe(true);
+
+    await waitFor(() => expect(mockGraphQLWSClient.subscribe.subscriptions).toHaveLength(2));
+
+    const subscription1 = mockGraphQLWSClient.subscribe.subscriptions[0];
+    const subscription2 = mockGraphQLWSClient.subscribe.subscriptions[1];
+
+    subscription1.push({
+      data: {
+        users: {
+          edges: [
+            {
+              node: {
+                id: "1",
+                email: "user1@test.com",
+              },
+            },
+          ],
+        },
+      },
+      revision: 1,
+    } as any);
+
+    subscription2.push({
+      data: {
+        users: {
+          edges: [
+            {
+              node: {
+                id: "2",
+                email: "user2@test.com",
+              },
+            },
+          ],
+        },
+      },
+      revision: 1,
+    } as any);
```

Comment:
add a third push to the first subscription after this one to exercise out of order pushing